resources/com/github/coreos/job.json

```diff
@@ -0,0 +1,52 @@
+{
+  "kind": "Template",
+  "apiVersion": "v1",
+  "metadata": {
+    "name": "clean-up",
+    "annotations": {
+      "openshift.io/display-name": "Clean Up Job",
+      "description": "Clean Up Job",
+      "template.openshift.io/bindable": "false"
+    }
+  },
+  "objects": [
+  {      
+  "apiVersion": "batch/v1",
+  "kind": "Job",
+  "metadata": {
+    "name": "${NAME}"
+  },
+  "spec": {
+    "ttlSecondsAfterFinished": 10800,
```

Comment:
If this isn't fully supported yet, let's go with the other option for now? (A sleeper pod, and then some code to delete exited sleeper pods.)

Reply:
I think it is just a matter to enable it in the cluster. According to https://kubernetes.io/docs/concepts/workloads/controllers/ttlafterfinished/: `this feature is currently alpha, and can be enabled with both kube-apiserver and kube-controller-manager feature gate TTLAfterFinished.`  I was hoping we could maybe open a ticket to the team responsible for the cluster to see if it is possible to be added.

Reply:
I'm not sure how easy it is to enable that feature in OCP. We can do some digging, though based on the docs, it seems like it'd require overriding the `kube-apiserver` and `kube-controller-manager` CLIs to have it enabled. I found https://docs.openshift.com/container-platform/4.5/nodes/clusters/nodes-cluster-enabling-features.html related to this, though there's a big caveat that upgrades are not officially supported.

Reply:
So overall, I would probably stick with the pod hack for now.

Or... if we have to run code *anyway* to delete the pod, I wonder if we could just not use ownerReferences at all for now and make the GC code query specifically for BCs which e.g. were created 24h ago?

Reply:
And then in a future release if we have the TTL bit stabilized, we can circle back and clean this up so we don't have to explicitly GC.

Reply:
So, I basically set 2h time for the job to complete, 2h there  is a plenty of time for the build to complete. After this time the job will be in complete state. The cleanup will look for job names starting with cleanup and will delete these in complete state. If a build fail the job will also end after 2h. Let me know if you want me to change something here =)